### ISSUE TITLE
Address Charts, Apps and Marketplace, and Cluster Tools Experience for RKE2 clusters with Windows Worker Nodes

### ISSUE THREAD

**Issue by rosskirkpat** [rancher/dashboard#3974]
Old title: `v2 Monitoring and Logging cannot be scheduled on Windows nodes in RKE2 custom clusters`

rancher issue: https://github.com/rancher/rancher/issues/34405

Related dashboard issues:

https://github.com/rancher/dashboard/issues/4983
https://github.com/rancher/dashboard/issues/4859
https://github.com/rancher/dashboard/issues/4986

**Rancher Server Setup**
- Rancher version: v2.6-head
```
Versions
Component 	Version
Rancher 	v2.6-be960478b7d8f83eecd3d33fbd8bdd2504ed409f-head
Helm 	v2.16.8-rancher1
Machine 	v0.15.0-rancher68
```
- Installation option (Docker install/Helm Chart): docker
- Proxy/Cert Details: self-signed

**Information about the Cluster**
- Kubernetes version:  v1.21.3-rc7+rke2r2
- Cluster Type (Local/Downstream): downstream
   - If downstream, what type of cluster? (Custom/Imported or specify provider for Hosted/Infrastructure Provider): Custom RKE2 

**Describe the bug**
When deploying v2 monitoring and logging apps, the Windows workloads are not being scheduled out of the box.

**To Reproduce**
Provision custom cluster with 1x linux all role or 1x etcd/cp and 1x worker
Enable v2 Logging and/or v2 monitoring apps with default helm chart configuration.

**Result**
The linux logging workload deploys as expected. There are no windows logging (fluentd) workloads deployed.

**Expected Result**
Windows workload (`rancher-logging-win-agent-fluentbit`) for v2 logging should be scheduled on Windows nodes.
Windows workload (`rancher-monitoring-windows-exporter`) for v2 monitoring should be scheduled on Windows nodes.

**Additional context**

Rancher and/or the UI need logic to detect that a windows node is present in a cluster and therefore set the `global.cattle.windows.enabled` value to `true` in the cluster.

Both the v2 monitoring and v2 logging charts rely on `.Values.global.cattle.windows.enabled` to be present and set to true.
https://github.com/rancher/charts/blob/dev-v2.6/charts/rancher-logging/rancher-logging/100.0.0%2Bup3.12.0/templates/_helpers.tpl#L68-L71

For v2 logging, the value was not present in the chart values yaml after a successful deployment of v2 logging.

For v2 monitoring, the value was set to false (below) in the chart values yaml after a successful deployment of v2 monitoring.
```
global:
  cattle:
    systemDefaultRegistry: ''
    windows:
      enabled: false
```

**Workaround**

Editing the values yaml for both v2 monitoring and v2 logging and setting `global.cattle.windows.enabled` to `true` allow the workloads to be scheduled.

```
global:
  cattle:
    windows:
      enabled: true
```

**Comment by vincent99** [comment]
The `...windows.enabled` flag is set based on the `management.cattle.io.cluster`'s `status.provider` being set to `rke.windows`, while rke2 is just `rke2`.  Need something in the cluster to key off of that doesn't require having access to see the nodes/machines/machinePools.

**Comment by phillipsj** [comment]
I think we should have a meeting to discuss this and make sure we are all on the same page on the solution, whatever that should be.

**Comment by gaktive** [comment]
@phillipsj let me and Vince know when it's time to meet over this.

**Comment by phillipsj** [comment]
@sirredbeard @rosskirkpat FYI.  Let me find a tim that works for all of us.

**Comment by phillipsj** [comment]
@deniseschannon FYI

**Comment by gaktive** [comment]
As discussed, UI will leverage how the providers for RKE1 fuel into the Dashboard and do the same for RKE2. We'll keep an eye on https://github.com/rancher/kubernetes-provider-detector/pull/8 and potential other PRs tied to the affiliated `rancher/rancher` ticket but Vince knows what to do.

**Comment by vincent99** [comment]
Theoretical UI support once the status updates:  https://github.com/rancher/dashboard/pull/3994

**Comment by sirredbeard** [comment]
@gaktive What are you we waiting on from backend for this? 

@luthermonson provided code that should have unblocked this via stats.

**Comment by gaktive** [comment]
Upon discussion with @luthermonson, the linked-to `rancher/windows` ticket is the sorta epic used to track this work, which now indicates that they are waiting on UI to complete this before ramping up so we're now blocking them. 

If UI is still stuck, then we would need to explicitly state why. The API should have the appropriate code to allow us to filter whether a system is Windows or not now.

@sirredbeard, does https://github.com/rancher/rancher/issues/35595 need to get in too?

**Comment by sirredbeard** [comment]
Ideally yes.

**Comment by rosskirkpat** [comment]
Related issues:

https://github.com/rancher/dashboard/issues/4983
https://github.com/rancher/dashboard/issues/4859

**Comment by mantis-toboggan-md** [comment]
@sirredbeard I think we need a little more clarity around exactly what behavior is desired on the UI side before proceeding with this issue - we now have the ability to detect whether or not an RKE2 cluster contains windows nodes via `status.windowsWorkerCount` but what are we doing with this information within the chart install process? I see an `rke2Windows.enabled` field was proposed and rejected: are we reverting to toggling `global.cattle.windows.enabled`?

cc @aiyengar2

**Comment by gaktive** [comment]
This should be addressed as part of comments per https://github.com/rancher/rancher/issues/36401#issuecomment-1049209643 so with all the other work that the UI is doing with RKE2 Windows, this may be irrelevant now.

Passing to QA in case there's anything they need to look at here but the key UI ticket for this work is https://github.com/rancher/dashboard/issues/5137 so that takes precedence.

**Comment by phillipsj** [comment]
@bmdepesa this one is blocked by the Cluster Tools issue.

**Comment by luthermonson** [comment]
This issue is going to be superceded by a previous dashboard issue which captures everything and has all PRs linked up. You can see context https://github.com/rancher/dashboard/issues/5137#issuecomment-1062323770
